epochStart/bootstrap/sovereignEpochStartBootstrapperFactory.go

```diff
@@ -20,19 +20,29 @@ func NewSovereignEpochStartBootstrapperFactory(esbf EpochStartBootstrapperCreato
 	}, nil
 }
 
-// CreateEpochStartBootstrapper creates a new epoch start bootstrapper for chain run sovereign
+// CreateEpochStartBootstrapper creates a new epoch start bootstrapper for sovereign chain operations
 func (bcf *sovereignEpochStartBootstrapperFactory) CreateEpochStartBootstrapper(epochStartBootstrapArgs ArgsEpochStartBootstrap) (EpochStartBootstrapper, error) {
-	epochStartBootstrapper, err := NewEpochStartBootstrap(epochStartBootstrapArgs)
+	return bcf.createSovereignEpochStartBootStrapper(epochStartBootstrapArgs)
+}
+
+// CreateStorageEpochStartBootstrapper creates a new storage epoch start bootstrapper for sovereign chain operations
+func (bcf *sovereignEpochStartBootstrapperFactory) CreateStorageEpochStartBootstrapper(epochStartBootstrapArgs ArgsStorageEpochStartBootstrap) (EpochStartBootstrapper, error) {
+	esb, err := bcf.createSovereignEpochStartBootStrapper(epochStartBootstrapArgs.ArgsEpochStartBootstrap)
 	if err != nil {
 		return nil, err
 	}
 
-	sesb, err := NewSovereignChainEpochStartBootstrap(epochStartBootstrapper)
+	epochStartBootstrapArgs.EpochStartBootStrap = esb.epochStartBootstrap
```

Comment:
This mutates the input `args` / side effect (the alternative would have been to clone it, then mutate it).

Reply:
Oh, now I see, `epochStartBootstrapArgs` is passed by value, all good :+1: